src/api/version.ts

```diff
@@ -163,7 +176,151 @@ export namespace VersionHelper {
+      return data.version
+    } catch {
+      return
+    }
+  }
+
+  /**
+   * Checks whether there is an update available for current chectl.
+   */
+  export async function isChectlUpdateAvailable(cacheDir: string, forceRecheck = false): Promise<boolean> {
+    // Do not use ctx inside this function as the function is used from hook where ctx is not yet defined.
+
+    if (getProjectName() !== 'chectl') {
+      // Do nothing for chectl flavors
+      return false
+    }
+
+    const currentVersion = getProjectVersion()
+    if (currentVersion === CHECTL_DEVELOPMENT_VERSION) {
+      // Skip it, chectl is built from source
+      return false
+    }
+
+    const channel = currentVersion.includes('next') ? 'next' : 'stable'
+    const newVersionInfoFilePath = path.join(cacheDir, `${channel}-${UPDATE_INFO_FILENAME}`)
+    let newVersionInfo: NewVersionInfoData = {
+      latestVersion: '0.0.0',
+      lastCheck: 0,
+    }
+    if (fs.existsSync(newVersionInfoFilePath)) {
+      newVersionInfo = JSON.parse(fs.readFileSync(newVersionInfoFilePath, 'utf8')) as NewVersionInfoData
+    }
+
+    const now = Date.now()
+    if (forceRecheck || now - newVersionInfo.lastCheck > A_DAY_IN_MS) {
+      // Cached info is expired. Fetch actual info about versions.
+      // undefined cannot be returned from getLatestChectlVersion as 'is flavor' chack was done before.
+      const latestVersion = (await getLatestChectlVersion(channel))!
+      newVersionInfo = { latestVersion, lastCheck: now }
+      fs.writeFileSync(newVersionInfoFilePath, JSON.stringify(newVersionInfo), { encoding: 'utf8' })
+    }
+
+    return compareVersions(newVersionInfo.latestVersion, currentVersion) > 0
+  }
+
+  /**
+   * Indicates if stable version of Eclispe Che is specified.
+   */
+  export function isStableVersion(flags: any): boolean {
+    if (flags.version === 'next' || flags.version === 'nightly') {
+      return false
+    }
+    return true
+  }
+
+  /**
+   * Indicates if the newest stable or nightly version is specified by its alias.
+   */
+  export function isTopVersion(flags: any): boolean {
+    return !flags.version || flags.version === 'next' || flags.version === 'nightly' || flags.version === 'latest' || flags.version === 'stable'
+  }
+
+  /**
+   * Removes 'v' prefix from version string.
+   * @param version version to process
+   * @param checkForNumber if true remove prefix only if a numeric version follow it (e.g. v7.x -> 7.x, vNext -> vNext)
+   */
+  export function removeVPrefix(version: string, checkForNumber = false): string {
+    if (version.startsWith('v') && version.length > 1) {
+      if (checkForNumber) {
+        const char2 = version.charAt(1)
+        if (char2 >= '0' && char2 <= '9') {
+          return version.substr(1)
+        }
+      }
+      return version.substr(1)
+    }
+    return version
+  }
+
+  /**
+   * Compares versions in format: x.y.z.w...
```

Comment:
looks like we should use https://www.npmjs.com/package/semver

Reply:
You are probably right. Deleted our util method and used `semver` lib instead.